### ISSUE TITLE
Database connection options ignored and health check not helpful

### ISSUE THREAD

**Issue by mattrobinsonsre** [keycloak/keycloak#19204]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

core

### Describe the bug

For HA purposes I have an installation of keycloak connected to a postgresql read replica and I have the db url configured with `target_session_attrs=primary`.
A connection with that configuration should fail on a read-replica and I have confirmed that it does fail using the same connection options to `psql`.  What actually happends is that the app starts normally and database health check returns 200 with no hint that it is aware of a database connection issue.
It seems clear to me that the database connection writeable check is not being performed as the app appears to function entirely normally, until you attempt to perform a write operation from the UI which inevitably fails.

### Version

20.0.3

### Expected behavior

A postgres database connection to a read replica with `target_session_attrs=primary` in the db url should fail.

### Actual behavior

The connection succeeds and the service reports healthy even though it cannot write to the database.

### How to Reproduce?

Connect the application to a postgres backend with either a user with read-only access or just connect it to a read replica.

**Comment by mposolda** [comment]
Moving to "store" as this looks like DB related issue.

**Comment by hmlnarik** [comment]
@mattrobinsonsre The Postgres parameters need to be passed in via `options` parameter or an alternative specific to JDBC driver which seems to be `targetServerType`, see https://jdbc.postgresql.org/documentation/use/#connection-parameters.

If this does not help, please share the full url (excluding the sensitive details).

**Comment by mattrobinsonsre** [comment]
Okay I mean I've looked at the code and the documentation and I've tried specifying the database connection options rather than the full uri and it makes no difference.
I think that the problem is that the quarkus library is overwriting any commands specified for execution on initial connection and the connection option `target_session_attrs` relies on executing test operations after initial connection.
quarkus allows these commands to be configured but there is no option in keycloak to pass this configuration through to quarkus.
At any rate we've solved our problem by scaling in the cold region so we don't have to rely on the broken health check logic.
